### PR TITLE
Decorator pattern

### PR DESCRIPTION
**Pull request by melaniesigrid**
Thank you for reviewing my project! On this PR, I...

- Created the base decorator nameable, and inheriting decorators. Make person inherit from Nameable, added methods to capitalize and trim string in decorators.

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  def correct_name
+    @nameable.correct_name.upcase
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? @nameable.correct_name.slice(0, 10) : @nameable.correct_name
+  end
+end
```
Comment (SajjadAhmad14) [inline decorator.rb]: Please move these two classes to their separate files. This way the project files are managed and the code is clean.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Melanie, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [x] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: decorator.rb (hunk 1 above)
Comment (melaniesigrid) [reply decorator.rb]: Thank you for your review! I have moved each decorator to its own file to better manage the project.

**Review by Emmyn5600 — APPROVED** [review]
Hi @melaniesigrid 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada: :clap: :clap: :100:

nodding-yes

## Optional suggestions

- [x] N/A

##### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

##### Cheers and Happy coding!:clap::clap::clap:
##### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**